### PR TITLE
Infra 531

### PR DESCRIPTION
**Pull request by sterlingwellscaffeine**
## Purpose
1.) Adds the Bifrost JSON-RPC routes as well as tests that will work so long as the local environment is setup properly

## Approach

1.) Munge the inputs to the correct format
2.) Send and format (if necessary) the request
3.) Format the output
Profit

## Testing

Lots of unit tests

## Tickets

INFRA-521

### CODE REVIEW

**Review by scasplte2 — APPROVED** [review]
In general I would like to see more comments throughout your code

File: brambl/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import sys
+
+import pkg_resources
+
+if sys.version_info < (3, 5):
+    raise EnvironmentError(
+        "Python 3.5 or above is required. ")
```
[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/__init__.py:7]: Readme would be a good idea. I will add that for the final product for sure.

File: brambl/base58/__init__.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
 
 from brambl.typing.encoding import Base58Str
 from brambl.utils.types import is_string
 
 
+def isBase58(sb: Union[str, bytes]) -> bool:
+    try:
+        if isinstance(sb, str):
+            # If there's any unicode here, an exception will be thrown and the function will return false
+            sb_bytes = bytes(sb, 'latin-1')
```
[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/base58/__init__.py:13]: I'm not assuming latin-1 encoding so I get the bytes to make sure that it is latin-1. That is the validator.

[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/base58/__init__.py:13]: I am not assuming a latin-1 encoding for this generic helper method.  The logic above is how I am validating that the string is Latin-1 encoded.

File: brambl/ed25519/datatypes.py [hunk 3]
```diff
@@ -203,8 +203,10 @@ def __init__(self,
         self._raw_key = key
         super().__init__()
 
-    def to_address(self, network_prefix: NetworkId, proposition_type: str) -> Address:
-        return public_key_bytes_to_address(self.to_bytes(), network_prefix, proposition_type)
+    def to_address(self, network_prefix: NetworkId, proposition_type: str) -> \
```
[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/ed25519/datatypes.py:206]: I see, what do you have in mind in terms or organization or address generation? I am open to reorganizing, but given that I was only tasked to implement with the PubKeyHash proposition type, that is what I am focused on here.

File: brambl/modules/exceptions.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from ... import exceptions
+
+
+class WalletRESTException(exceptions.BackendException, exceptions.WalletException):
```
[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/modules/exceptions.py:4]: I will update the language to reflect that, thanks!

File: brambl/types.py [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
+from brambl.ed25519.utils.address import Address
+from brambl.typing.encoding import Base58Str, HexStr
 
 TReturn = TypeVar("TReturn")
 RPCEndpoint = NewType("RPCEndpoint", str)
 
+if TYPE_CHECKING:
+    from brambl import Brambl  # noqa: F401
+
+Poly = NewType('Poly', int)
```
[the remark this answers is not in the document]
Comment (sterlingwellscaffeine) [reply brambl/types.py:13]: Python ints are unbounded

**Pull request merged**